Makefile

```diff
@@ -58,3 +63,19 @@ shellcheck:
 
 test-melange: $(KEY)
 	$(MELANGE) test --runner=docker melange.yaml $(MELANGE_OPTS) $(MELANGE_TEST_OPTS)
+
+.PHONY: lint
+lint: yam-check shellcheck
+
+.PHONY: yam-check yam
+# yam-check shows changes it would make and exits 0 on no changes.
+yam-check: $(BIN_TOOLS_D)/yam
+	$(BIN_TOOLS_D)/yam --lint $(YAM_FILES)
+
+# yam applies changes to the files you cannot trust its exit code
+yam: $(BIN_TOOLS_D)/yam
+	$(BIN_TOOLS_D)/yam $(YAM_FILES)
+
+$(BIN_TOOLS_D)/yam:
+	@mkdir -p $(BIN_TOOLS_D)
+	GOBIN=$(BIN_TOOLS_D) go install github.com/chainguard-dev/yam@v0.2.29
```

Comment:
I think we can consider approaching it differently reason being once `yam` gets installed it'll always say the following

```bash
make: `/foo/bar/tw/tools/bin/yam' is up to date.
```

we can define a top-level variable here and copy that file inside `tools/bin` say `./tools/bin/yam-version.txt` and if that drifts from the value of the variable `yam-version` that we define in Makefile then we `rm ./tools/bin/yam` and install it again.